### PR TITLE
Viite 2371 handle roadway numbers in two track cases

### PR DESCRIPTION
**Pull request by andre-policarpo**
- Master branch was merged today (30/07)
- All the tests are passing, [here](http://livibuild04.vally.local/jenkins/job/viite/job/Run-Tests/405/consoleFull).
- Additional comments to the US Jiira's ticket will be added only tomorrow probably.
- Changes after Sami's Code Review were also pushed - just some commented code properly removed.

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala [hunk 1]
```diff
@@ -81,10 +81,10 @@ case class ProjectLink(id: Long, roadNumber: Long, roadPartNumber: Long, track:
   }
 
   def addrMLength(): Long = {
-    if (isSplit)
+//    if (isSplit)
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala:84]: Remove commented out code

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectSectionMValueCalculator.scala [hunk 2]
```diff
@@ -77,13 +77,15 @@ object ProjectSectionMValueCalculator {
     }
   }
 
-  def assignLinkValues(seq: Seq[ProjectLink], addrSt: Long): Seq[ProjectLink] = {
+  def assignTerminatedLinkValues(seq: Seq[ProjectLink], addrSt: Long): Seq[ProjectLink] = {
     val newAddressValues = seq.scanLeft(addrSt) { case (m, pl) =>
       pl.status match {
         case LinkStatus.Terminated =>
-          if(pl.isSplit) m + pl.endAddrMValue - pl.startAddrMValue else m + pl.addrMLength
+//          if (pl.isSplit) m + pl.endAddrMValue - pl.startAddrMValue else
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/ProjectSectionMValueCalculator.scala:84]: Remove commented out code

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/TrackSectionRoadway.scala [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+        (continuousSection :+ seq.drop(continuousSection.size).head, seq.drop(continuousSection.size + 1))
+      else
+        (continuousSection, seq.drop(continuousSection.size))
+    }
+
+    val assignedContinuousSection = assignRoadwayNumbersInContinuousSection(continuousProjectLinks, givenRoadwayNumber)
+    (assignedContinuousSection, restProjectLinks)
+  }
+
+  private def assignRoadwayNumbersInContinuousSection(links: Seq[ProjectLink], givenRoadwayNumber: Long): Seq[ProjectLink] = {
```
Comment (skosonen) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/TrackSectionRoadway.scala:73]: Do we assume here that links are always in right order?

Comment (andre-policarpo) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/TrackSectionRoadway.scala:73]: Yes, for every process regarding addresses and roadway numbers, project links must be ordered by _startAddrMValue_.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala (hunk 1 above)
Comment (andre-policarpo) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala:84]: Done.

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala [hunk 4]
```diff
@@ -2015,8 +2015,6 @@ class ProjectServiceSpec extends FunSuite with Matchers with BeforeAndAfter {
       projectService.updateProjectLinks(savedProject.id, Set(), linkIds207.toSeq, LinkStatus.Transfer, "-", 5, 207, 0, Option.empty[Int]) should be(None)
       projectService.updateProjectLinks(savedProject.id, Set(), Seq(5168510), LinkStatus.Terminated, "-", 5, 207, 0, Option.empty[Int]) should be(None)
```
Comment (skosonen) [inline digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala:2016]: 5168510 id could be extracted as own value.

**Pull request merged**